### PR TITLE
Make scope an (optional) CLI arg

### PR DESCRIPTION
**Pull request by shayne-fletcher**
Scope can now be provided as a command line argument with default `@daml.js` (as requested in https://github.com/digital-asset/daml/pull/4799#discussion_r387885385).

### CODE REVIEW

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 1]
```diff
@@ -89,13 +90,19 @@ optionsParser = Options
...
         )
     <*> optional (strOption
         (  short 'p'
         <> metavar "PACKAGE-JSON"
         <> help "Path to an existing 'package.json' to update"
         ))
+    <*> (Scope <$> strOption
+        (  short 's'
+        <> metavar "SCOPE"
+        <> value "@daml.js"
```
Comment (cocreature) [inline language-support/ts/codegen/src/TsCodeGenMain.hs]: What’s the point of including the `@` here? Can a scope ever not contain an `@`? My understanding was that the `@` sign is not part of the scope name but just syntax to differentiate the scope from the name. So it seems more sensible to not include it here.

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 2]
```diff
@@ -661,22 +661,22 @@ instance ToJSON PackageJson where
...
-setupWorkspace optOutputDir dependencies file = do
+setupWorkspace optOuputDir dependencies file = do
   let (g, nodeFromVertex) = graphFromEdges'
         (map (\(a, ds) -> (a, a, map unDependency ds)) dependencies)
       ps = map (fst3 . nodeFromVertex) $ reverse (topSort g)
         -- Topologically order our packages.
-      scope = scopeOfScopeDir optOutputDir
-        -- 'scope' we expect to be something like "daml2ts".
-  let ourWorkspaces = map ((unScope scope <> "/") <>) ps
+      scopeDir = T.pack $ takeFileName optOuputDir
```
Comment (cocreature) [inline language-support/ts/codegen/src/TsCodeGenMain.hs]: Does anything break if I set `optOutputDir` to something different than the scope? Maybe a good idea to have a testcase for that?

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 3]
```diff
@@ -661,22 +661,22 @@ instance ToJSON PackageJson where
...
+        -- The leaf directory of the output directory (e.g. often 'daml2ts').
```
Comment (cocreature) [inline language-support/ts/codegen/src/TsCodeGenMain.hs]: It looks like this is the root not the leaf. Am I missing something?

File: language-support/ts/codegen/src/TsCodeGenMain.hs (hunk 1 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: @cocreature in https://github.com/digital-asset/daml/pull/4818#discussion_r388145444, @hurryabit indicated that he'd prefer to see the '@'. i'm easy on this point.

File: language-support/ts/codegen/src/TsCodeGenMain.hs (hunk 2 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: There is no relationship between scope and the output directory. For example, the DAVL unit test defaults the scope to `@daml.js` but the output directory is `daml2ts`.

File: language-support/ts/codegen/src/TsCodeGenMain.hs (hunk 3 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: No it's the leaf. We take the filename of `/some/long/path/to/here` giving 'here'.

File: language-support/ts/codegen/src/TsCodeGenMain.hs (hunk 1 above)
Comment (cocreature) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: That was about the internals not the UX for the CLI argument afaict. Either way, if @hurryabit prefers this I don’t feel too strongly about this.

File: language-support/ts/codegen/src/TsCodeGenMain.hs (hunk 2 above)
Comment (cocreature) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: Then maybe this should not be called `scopeDir` given that it’s completely independent of the scope?

File: language-support/ts/codegen/src/TsCodeGenMain.hs (hunk 3 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: Wow. Is it really spelled `optOuputDir` in the code? Better fix that :)

Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: Fixed.

Comment (cocreature) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: oh i see, I misread the code. Thanks for the explanation!

File: language-support/ts/codegen/src/TsCodeGenMain.hs (hunk 2 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: well, it's the scope dir in that it's the directory that contains all the packages that are associated with the scope

Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: i could rename it to say, `leafDir` or `outDir` or something if you think it confusing?

Comment (cocreature) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: Maybe `outBaseName` or something like that?

Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: ok. will do.

File: language-support/ts/codegen/src/TsCodeGenMain.hs (hunk 1 above)
Comment (hurryabit) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: As @cocreature said, I preferred it for the internals. If we required our users to pass the `@` to the CLI argument, we also need to check that they actually do it, which we currently don't. That said, I prefer if our would not need to pass in the `@` symbol but just `daml.js`.

Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: https://github.com/digital-asset/daml/pull/4939

**Pull request merged**